cmd/crank/init.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/alecthomas/kong"
+	"github.com/go-git/go-billy/v5/osfs"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/storage/memory"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+)
+
+// initCmd initializes a repository from a template
+type initCmd struct {
+	Function initFunctionCmd `cmd:"" help:"Initialize a Function from the repo template for a selected language."`
+}
+
+type initFunctionCmd struct {
+	Name string `arg:"" help:"Name of the Function to initialize."`
+
+	Language  functionLanguage `short:"l" help:"Language of the Function to initialize." enum:"go" default:"go"`
+	Directory string           `short:"d" help:"Path of the directory to initialize." default:"." type:"path"`
+}
+
+type functionLanguage string
+
+const (
+	functionLanguageGo functionLanguage = "go"
+)
+
+func (c *initFunctionCmd) GetURLForLanguage() (string, error) {
+	switch c.Language {
+	case functionLanguageGo:
+		return "https://github.com/crossplane/function-template-go", nil
+	default:
+		return "", errors.Errorf("unknown language %s", c.Language)
+	}
+}
```

Comment:
If we think of this as a "template" rather than a language, I wonder if we could have a few well known templates then fall back to treating unknown values as a template repo.

Assuming this becomes an argument (as I suggested in another comment) this could work like:

```shell
# Initialize a Go function
$ crossplane init function go

# Initialize a Python function
$ crossplane init function python

# Initialize a Function using some bespoke template
$ crossplane init function https://github.com/negz/my-special-fork-of-function-template-go
```

Reply:
Sounds good to me moving to an argument. However I'm not sure there is much value in having it work with random urls, there is nothing special about what we are doing, if someone wants to use a custom template folder they can just clone the repo and copy stuff  over manually.

Reply:
I think it would be useful to support custom URLs, especially if we end up taking care of renaming some template values to what you actually want your Function to be called.

Reply:
I did it as a flag, I forgot to update the description